pkg/nats/proxy/compute_proxy.go

```diff
@@ -27,11 +27,14 @@ type ComputeProxyParams struct {
 // is provided.
 type ComputeProxy struct {
 	conn            *nats.Conn
-	streamingClient *stream.Client
+	streamingClient *stream.ConsumerClient
 }
 
 func NewComputeProxy(params ComputeProxyParams) (*ComputeProxy, error) {
-	sc, err := stream.NewClient(stream.ClientParams{Conn: params.Conn})
+	sc, err := stream.NewConsumerClient(stream.ConsumerClientParams{
+		Conn:                params.Conn,
+		HeartBeatRequestSub: requesterEndpointPublishSubject(params.Conn.Opts.Name, StreamHeartBeat),
```

Comment:
I see your point:
- Having a deterministic subject will allow the compute node to fail faster when the requester restarts (assuming the requester will restart within ~5 seconds)
- Whereas a random subject means the compute heartbeat will either timeout, or the nats server _might_ return a no active subscriber error. If there was a timeout, we shouldn't have the compute node drop the streamIDs immediately as it might a transient error. If nats server returns no active subscriber, then we can drop the streams immediately, but I doubt nats server will figure that fast enough

Both are valid options and feel free to use whatever you think is right. My concern would be about using `Conn.Opts.Name` as the subject identifier. I understand we are today using the nodeID as the connection name, but it is an optional field in nats and we might change things in the future if we were to allow multiple nats connections per node for different purposes